src/UNL/Knowledge/Records.php

```diff
@@ -124,4 +135,108 @@ protected function getMergedKey($item, $section, $tags)
 
     	return implode(' ', $results);
     }
+
+    protected function getPublicProperties()
+    {
+        $self = $this;
+        $getPublicProperties = function() use ($self) {
+            return get_object_vars($self);
+        };
+        $getPublicProperties = $getPublicProperties->bindTo(null, null);
+
+        return $getPublicProperties();
```

Comment:
I had to do some research to figure out what was happening here. It seems like something like `(new ReflectionObject($this))->getProperties(ReflectionProperty::IS_PUBLIC);` would be easier to read.